sympy/ntheory/generate.py

```diff
@@ -810,20 +833,25 @@ def primorial(n, nth=True):
     primerange : Generate all primes in a given range
 
     """
-    if nth:
-        n = as_int(n)
-    else:
-        n = int(n)
-    if n < 1:
-        raise ValueError("primorial argument must be >= 1")
-    p = 1
-    if nth:
-        for i in range(1, n + 1):
-            p *= prime(i)
-    else:
-        for i in primerange(2, n + 1):
-            p *= i
-    return p
+    @classmethod
+    def eval(cls, n, nth=True):
+        if isinstance(n, Symbol):
```

Comment:
The nth parameter is not considered here. You will need to make a symbolic adjustment so it is correct.

Reply:
I am not able to understand the need.
```
        if not n.is_Number:
            return None

        if nth:
            n = as_int(n)
        else:
            n = int(n)
```
I do not think we need the check separately. if nth is False and n is not an Integer, it will cause a ValueError.